### PR TITLE
feat: add default credential providers to all clients

### PR DESCRIPTION
**Pull request by anitarua**
Work towards https://github.com/momentohq/dev-eco-issue-tracker/issues/975

Added a new function `getDefaultCredentialProvider` to the core package that's used in both node and web sdks when creating clients with optional credential providers (should be all 5 of them now: cache, topics, auth, leaderboard, storage)

### CODE REVIEW

File: packages/client-sdk-nodejs/src/index.ts [hunk 1]
```diff
@@ -117,6 +117,7 @@ import {
   CredentialProvider,
   StringMomentoTokenProvider,
   EnvMomentoTokenProvider,
+  getDefaultCredentialProvider,
```
Comment (cprice404) [inline packages/client-sdk-nodejs/src/index.ts:120]: Do we intend for this to be a user-facing function? If not, I think we might not want to export it from the nodejs/web sdk packages.

File: packages/core/src/index.ts [hunk 2]
```diff
@@ -107,6 +107,7 @@ import {
   CredentialProvider,
   StringMomentoTokenProvider,
   EnvMomentoTokenProvider,
+  getDefaultCredentialProvider,
```
Comment (cprice404) [inline packages/core/src/index.ts:110]: it makes sense to export it here so the SDKs can use it.

**Review by cprice404 — COMMENTED** [review]
just one nit about exports

File: packages/client-sdk-nodejs/src/index.ts (hunk 1 above)
Comment (anitarua) [reply packages/client-sdk-nodejs/src/index.ts:120]: oh probably not, will remove it from there

**Review by cprice404 — APPROVED** [review]
default

**Pull request merged**